backend/src/apiserver/list/list.go

```diff
@@ -44,15 +44,22 @@ type token struct {
 	// SortByFieldValue is the value of the sorted field of the next row to be
 	// returned.
 	SortByFieldValue interface{}
+	// SortByFieldIsRunMetric indicates whether the SortByFieldName field is
+	// a run metric field or not.
+	SortByFieldIsRunMetric bool
```

Comment:
nit: do you forsee we will have other `SortByFieldIsXXXX`s?

Should we use an enum here?

Reply:
With our existing model (pipeline, pipeline version, run, experiment, job), we probably won't have further special field like run metrics. E.g., pipeline, pipeline version and experiment only have fields that are directly mapped to table columns. As for job, the only array-typed field (that is, field not mapped to table columns) is resource references. Resource reference value is categorical and we use filter on it. As for run, the only arrary-typed fields are run metrics and resource references.

In case we'll extend our existing models with more fields. We'll probably be more careful not to use a field that is not mapped to a column and then want to sort on that field. The sql query is not very efficient in the case. That being said, if there is an absolute need to do so, the dev who implements it would be responsible to add that field with all the sorting/filtering mechanism and/or refactoring the existing framework of sorting/filtering on our models.

Reply:
SGTM, thanks for the detailed explanation!